dashboard/app/models/user.rb

```diff
@@ -1955,21 +1959,16 @@ def pl_units_started
     pl_user_scripts = user_scripts.select {|us| us.script.pl_course?}
     pl_scripts = pl_user_scripts.map(&:script)
 
-    levels = pl_scripts.map(&:levels).flatten
-    level_ids = levels.map(&:id)
-
-    # Handle levels-within-levels
-    levels.each {|l| level_ids << l.contained_levels.first&.id unless l.contained_levels.empty?}
-
-    user_levels = UserLevel.where(user: self, script: pl_scripts, level_id: level_ids)
-    return [] if user_levels.empty?
-    user_levels_by_script = user_levels.group_by(&:script_id)
     percent_completed_by_script = {}
     pl_scripts.each do |pl_script|
-      levels_completed = (user_levels_by_script[pl_script.id] || []).count(&:passing?)
+      if pl_user_scripts.find {|us| us.script_id == pl_script.id}.completed_at
```

Comment:
I've added this as an optimization more than anything. The code that controls the "Print Certificate" button on the Unit Overview page is [here](https://github.com/code-dot-org/code-dot-org/blob/d16707537f8911e20ab119c5d3532e58eda2279b/dashboard/lib/policies/script_activity.rb#L7) and does use `user_script`